### ISSUE TITLE
Diagnostic for an unused import from a macro expansion mentions a metavariable instead of the imported name

### ISSUE THREAD

**Issue by PatchMixolydic** [rust-lang/rust#103856]
Given the following code ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=e76e4cbfb0443c9d3a4f12f38a65ea6e)):

```rust
#[macro_export]
macro_rules! pub_mac {
    (pub(crate) macro_rules! $name:ident $body:tt) => {
        #[allow(unused_macros)]
        macro_rules! $name $body

        pub(crate) use $name;
    };
}

mod foo {
    crate::pub_mac! {
        pub(crate) macro_rules! bar {
            () => {};
        }
    }
}
```

The current output is:

```
warning: unused import: `$name`
  --> src/lib.rs:7:24
   |
7  |           pub(crate) use $name;
   |                          ^^^^^
...
12 | /     crate::pub_mac! {
13 | |         pub(crate) macro_rules! bar {
14 | |             () => {};
15 | |         }
16 | |     }
   | |_____- in this macro invocation
   |
   = note: `#[warn(unused_imports)]` on by default
   = note: this warning originates in the macro `crate::pub_mac` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Ideally the output should look like:

```
warning: unused import: `bar`
  --> src/lib.rs:7:24
   |
7  |           pub(crate) use $name;
   |                          ^^^^^
...
12 | /     crate::pub_mac! {
13 | |         pub(crate) macro_rules! bar {
14 | |             () => {};
15 | |         }
16 | |     }
   | |_____- in this macro invocation
   |
   = note: `#[warn(unused_imports)]` on by default
   = note: this warning originates in the macro `crate::pub_mac` (in Nightly builds, run with -Z macro-backtrace for more info)
```

@rustbot modify labels +A-macros
